src/openms_gui/include/OpenMS/VISUAL/LayerDataIdent.h

```diff
@@ -60,7 +60,26 @@ namespace OpenMS
     /// move assignment
     LayerDataIdent& operator=(LayerDataIdent&& ld) = default;
 
-    std::unique_ptr<Painter1DBase> getPainter1D() const override;
+    std::unique_ptr<Painter2DBase> getPainter2D() const override;
+
+    std::unique_ptr<LayerData1DBase> to1DLayer() const override
+    {
+      throw Exception::NotImplemented(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
+    }
+
+    std::unique_ptr<LayerStoreData> storeVisibleData(const RangeAllType& visible_range, const DataFilters& layer_filters) const override;
+
+    std::unique_ptr<LayerStoreData> storeFullData() const override;
+
+    ProjectionData getProjection(const DIM_UNIT unit_x, const DIM_UNIT unit_y, const RangeAllType& area) const override;
+
+    PeakIndex findHighestDataPoint(const RangeAllType& area) const override
+    { // todo: not implemented
+      return PeakIndex();
```

Comment:
Nope, because this will be called for all layers. If a layer does not support it, it returns a defaulted (i.e. invalid) PeakIndex and nothing gets hightlighted.
Throwing an exception would lead to a message window every time you move the mouse :)